proxy/scripts/http_protocol_extractor.py

```diff
@@ -1,28 +1,67 @@
 import datetime
-import os
+import hashlib
 import json
-
+import os
 import urllib
+from urllib import parse
+
 from mitmproxy import ctx
 
 
 class HttpProtocolExtractor:
     def __init__(self):
         self.request_protocol = {}
+        self.hashes = []
+        self.request_count_unique = 0
+        self.request_count = 0
```

Comment:
Why are we introducing hashes here? Can we use a list for the requests and just output the length of the list? Then we can construct a set from the list for the unique values.

Reply:
We need hashes as that a combination of url, parameters and other methods. 
This make a unique request.

We count the number of actual request that get through the proxy and the number of unique hashes that we identify

Reply:
This is duplication from the alternate server replay script. Could we track the requests there, and access the values from here? Perhaps one addon can depend on another? Otherwise, we could make them both depend on another module to reduce the duplication.

Reply:
I agree with you here.

As this will modify the way proxy works can we do it in a separate change request. 
I will need further  testing and investigation

Reply:
https://github.com/mozilla/raptor-studio/issues/21

Reply:
https://github.com/mozilla/raptor-studio/issues/22